### PR TITLE
Offload OpenCL GPU testing to AWS EC2 as backup

### PR DESCRIPTION
**Pull request by serban-nicusor-toptal**
## Summary

We're relying on a physical machine for GPU testing and it's sadly offline, we're now trying to setup AWS EC2 GPU instances to use until everything gets back to normal.

## Tests

Describe the new tests with the pull request.

For bug fixes there should be a new test that would fail if the patch weren't in place (so that the bug could be caught if it comes up again).

For new features there should be at least one test showing the expected behavior and one test that demonstrates error handling. Be aware the reviewer will very likely ask for more tests than this, but it's a start.

## Side Effects

Are there any side effects that we should be aware of?

## Release notes

Replace this text with a short note on what will change if this pull request is merged in which case this will be included in the release notes.

## Checklist

- [ ] Math issue #(issue number)

- [ ] Copyright holder: (fill in copyright holder information)

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [ ] the code is written in idiomatic C++ and changes are documented in the doxygen

- [ ] the new changes are tested

### CODE REVIEW

File: Jenkinsfile
Comment (rok-cesnovar) [inline Jenkinsfile:246]: Mind adding the same for the Windows machine as well?

Comment (rok-cesnovar) [reply Jenkinsfile:246]: @serban-nicusor-toptal also we can change the label to "gelman-group-win2 || linux-gpu" and then add linux-gpu to the AWS instance and the gg-linux?

**Comment by rok-cesnovar** [comment]
@SteveBronder @bbbales2 hold off on restarting any other tests until this PR is merged. They wont pass.

File: Jenkinsfile
Comment (serban-nicusor-toptal) [reply Jenkinsfile:246]: Sure! This is actually a great idea to simplify matters.

**Comment by serban-nicusor-toptal** [comment]
Hey @rok-cesnovar I think we're good for now.

I had to hardcode:
- OPENCL_PLATFORM_ID=1
- OPENCL_DEVICE_ID=0

In the machine when it boots because it wasn't setup and I didn't figure out what should have set it up automatically. Since we're using a pool with 1GPU/instance it shouldn't cause any trouble I hope.
When this is done, I will add it `gg-linux` label so old PRs can finish and we can merge this in too.

File: Jenkinsfile
Comment (rok-cesnovar) [inline Jenkinsfile:274]: Sorry, I didnt think of this earlier. These should be 

```suggestion
                                sh "echo OPENCL_PLATFORM_ID=${env.OPENCL_PLATFORM_ID_GPU} >> make/local"
                                sh "echo OPENCL_DEVICE_ID=${env.OPENCL_DEVICE_ID_GPU} >> make/local"
```

File: Jenkinsfile
Comment (rok-cesnovar) [inline Jenkinsfile:283]: Same as above.

**Review by rok-cesnovar — COMMENTED** [review]
Lets leave the tests running now to see they pass. Then we should comment out CPU tests for now (we dont have instances with OpenCL CPU capabilites).

And also rename the env variables to have two sets:
${env.OPENCL_PLATFORM_ID_GPU}
${env.OPENCL_DEVICE_ID_GPU}
and
${env.OPENCL_DEVICE_ID_CPU}
${env.OPENCL_PLATFORM_ID_CPU}

Will make it easier to maintain.

**Comment by rok-cesnovar** [comment]
@serban-nicusor-toptal I think the platform id is 0.

**Comment by serban-nicusor-toptal** [comment]
Will change and retry.

**Review by rok-cesnovar — APPROVED** [review]
Looks great! Merge if tests pass. Thanks for fast action on this!

**Pull request merged**
